src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/EarthSlimeSlingItem.java

```diff
@@ -0,0 +1,44 @@
+package slimeknights.tconstruct.gadgets.item.slimesling;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.RayTraceContext;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.common.Sounds;
+import slimeknights.tconstruct.library.SlimeBounceHandler;
+
+public class EarthSlimeSlingItem extends BaseSlimeSlingItem {
+
+  public EarthSlimeSlingItem(Properties props) {
+    super(props);
+  }
+
+  /** Called when the player stops using an Item (stops holding the right mouse button). */
+  @Override
+  public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+    if (!(entityLiving instanceof PlayerEntity)) {
+      return;
+    }
+
+    PlayerEntity player = (PlayerEntity) entityLiving;
+    float f = getForce(stack, timeLeft);
+
+    // check if player was targeting a block
+    BlockRayTraceResult mop = rayTrace(worldIn, player, RayTraceContext.FluidMode.NONE);
+    if (mop.getType() == RayTraceResult.Type.BLOCK) {
```

Comment:
Needs the miss sound when its not targeting a block.

Might be worth adding a method to `BaseSlimeSlingItem`, something like `playSlingSound(PlayerEntity entity, boolean isSuccess)`